spring-boot/admin/src/main/resources/templates/admin/fragments/api-keys/edit.html

```diff
@@ -227,40 +236,37 @@ <h3> Select API user from table
                 // Add behaviour when the user saves the selection
                 $('#saveApiKeyButton').click(function () {
                     // Check if any github user selection
-                    if (table.row('.selected') !== null && table.row('.selected').data() !== null && typeof table.row('.selected').data() !== 'undefined')
-                    {
+                    if (table.row('.selected') !== null && table.row('.selected').data() !== null && typeof table.row('.selected').data() !== 'undefined') {
                         // Check if any service selection
-                        if ($("#selectedServiceId").val() !== null && $("#selectedServiceId").val() !== "" && typeof $("#selectedServiceId").val() !== 'undefined')
-                        {
-                            // Retrieve the github username which is common attribute in github user database and coatrack user database
-                            $("#selectedUserId").val(table.row('.selected').data().login);
+                        if ($("#selectedServiceId").val() !== null && $("#selectedServiceId").val() !== "" && typeof $("#selectedServiceId").val() !== 'undefined') {
+                            // Retrieve the github/keyrock username which is common attribute in github user database and coatrack user database
+                            ([[${@environment.getProperty('ygg.admin.idm.provider')}]].localeCompare("github") === 0) ? $("#selectedUserId").val(table.row('.selected').data().login) :
+                                $("#selectedUserId").val(table.row('.selected').data().username);
                             $("#apiKeyForm").submit();
-
-                        } else
-                        {
+                        } else {
                             $("#message").html("You must select a service");
                             $("#infoDialog").dialog("open");
                         }
-                    } else
-                    {
+                    } else {
                         $("#message").html("You must select a coatrack user");
                         $("#infoDialog").dialog("open");
                     }
                 });
             });
 
-            function searchGituhbUsers()
-            {
+            function searchGituhbUsers() {
                 $('#githubUserSearchButton').hide();
                 $('#busy_indicator').show();
 
-                table.ajax.url(encodeURI('/admin/github/search/' + $("#githubUserSearchCriteria").val())).load();
+                if ([[${@environment.getProperty('ygg.admin.idm.provider')}]].localeCompare("github") === 0) {
```

Comment:
You are doing this property check several times, which is quite difficult to read. Wouldn't it be easier to do this only one and put it into a "global variable" for this thymeleaf template? Or set some model attributes like `isGithubUsedAsIdm` and `isKeyrockUsedAsIdm` in the Java controller method?

Reply:
In order to facilitate this, I created a method called isGithubUsedAsIdm, which is always used to check if Github is being used as idm. I also shorten and compacted the code.
Based on this, there are no more repetitions